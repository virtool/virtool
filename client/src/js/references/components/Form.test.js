import ReferenceForm from "./Form";

describe("<ReferenceForm />", () => {
    let props = {
        state: {
            name: "test",
            description: "test-description",
            dataType: "test-datatype",
            organism: "test-organism",
            onChange: jest.fn(),
            errorName: "",
            errorDataType: ""
        }
    };
    let wrapper;

    it("renders correctly without errors", () => {
        props = {
            state: {
                ...props.state,
                errorFile: null,
                errorSelect: null
            }
        };
        wrapper = shallow(<ReferenceForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly with errors", () => {
        props = {
            state: {
                ...props.state,
                errorFile: null,
                errorSelect: "Error Select"
            }
        };
        wrapper = shallow(<ReferenceForm {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

});
