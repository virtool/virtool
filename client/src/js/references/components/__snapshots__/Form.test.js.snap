// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferenceForm /> renders correctly with errors 1`] = `
<div>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <div
        className="input-form-error"
      >
        <span
          className="input-error-message"
          style={
            Object {
              "margin": "0 0 0 0",
            }
          }
        >
          Error Select
        </span>
      </div>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <InputError
        error=""
        label="Name"
        name="name"
        value="test"
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <InputError
        label="Description"
        name="description"
        type="textarea"
        value="test-description"
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <InputError
        error=""
        label="Data Type"
        name="dataType"
        type="select"
        value="test-datatype"
      >
        <option
          className="text-capitalize"
          hidden={true}
          value=""
        />
        <option
          className="text-capitalize"
          hidden={false}
          key="genome"
          value="genome"
        >
          genome
        </option>
      </InputError>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <InputError
        label="Organism"
        name="organism"
        value="test-organism"
      />
    </Col>
  </Row>
</div>
`;

exports[`<ReferenceForm /> renders correctly without errors 1`] = `
<div>
  <Row
    bsClass="row"
    componentClass="div"
  />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <InputError
        error=""
        label="Name"
        name="name"
        value="test"
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <InputError
        label="Description"
        name="description"
        type="textarea"
        value="test-description"
      />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <InputError
        error=""
        label="Data Type"
        name="dataType"
        type="select"
        value="test-datatype"
      >
        <option
          className="text-capitalize"
          hidden={true}
          value=""
        />
        <option
          className="text-capitalize"
          hidden={false}
          key="genome"
          value="genome"
        >
          genome
        </option>
      </InputError>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={6}
      xs={12}
    >
      <InputError
        label="Organism"
        name="organism"
        value="test-organism"
      />
    </Col>
  </Row>
</div>
`;
